jw-player/include/shortcode.php

```diff
@@ -208,40 +209,71 @@ function jwplayer_shortcode_create_js_embed( $media_hash, $player_hash = null, $
 		$xml = "$xml?exp=$expires&sig=$signature";
 	}
 
-	$params = jwplayer_shortcode_filter_player_params($params);
+	$params = jwplayer_shortcode_filter_player_params( $params );
 	if ( count( $params ) ) {
 		// Support for player tracks.
-		foreach (array('sources', 'tracks') as $option) {
-			if ( isset($params[$option]) ) {
-				$json = '[' . $params[$option] . ']';
-				$obj = json_decode(preg_replace('/[{, ]{1}(\w+):/i', '"\1":', $json));
+		foreach ( array( 'sources', 'tracks' ) as $option ) {
+			if ( isset( $params[ $option ] ) ) {
+				$json = '[' . $params[ $option ] . ']';
+				$obj = json_decode( preg_replace( '/[{, ]{1}(\w+):/i', '"\1":', $json ) );
 				if ( null === $obj ) {
-					$json = str_replace(array('"',  "'"), array('\"', '"'), $json);
-					$obj = json_decode(preg_replace('/[{, ]{1}(\w+):/i', '"\1":', $json));
+					$json = str_replace( array( '"',  "'" ), array( '\"', '"' ), $json );
+					$obj = json_decode( preg_replace( '/[{, ]{1}(\w+):/i', '"\1":', $json ) );
 				}
-				$params[$option] = $obj;
+				$params[ $option ] = $obj;
 			}
 		}
 	}
-	if ( ! isset ( $params['source'] ) ) {
+	if ( ! isset( $params['source'] ) ) {
 		$params['playlist'] = $xml;
 	}
-	$paramstring = json_encode($params);
-	foreach (  array( '&amp;' => '&', '&#038;' => '&', '\/' => '/' ) as $from => $to ) {
-		$paramstring = str_replace($from, $to, $paramstring);
```

Comment:
yeah, sorry, this is a bit confusing since i added so much stuff in the lines below. 

i removed this, since the function wp_json_encode coverts the parameters array into a usable json format in your jwplayer js function